lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -152,6 +153,133 @@ class ConvertAtenFlipOp : public OpConversionPattern<AtenFlipOp> {
 };
 } // namespace
 
+// Returns updated output dimension and scale_factor converted to integer.
+static std::pair<Value, Value> getScaledDims(OpBuilder &builder, Location loc,
+                                        Value dim, double scaleFactor) {
+
+  Value scaleFactorInt = builder.create<arith::ConstantOp>(
+      loc, builder.getI64IntegerAttr((int64_t)ceil(scaleFactor)));
+  Value scaleFactorsFp = builder.create<arith::ConstantOp>(
+      loc, builder.getF64FloatAttr(scaleFactor));
+  Value dimInt = castIndexToInt64(builder, loc, dim);
+  Value dimFp =
+      builder.create<arith::SIToFPOp>(loc, scaleFactorsFp.getType(), dimInt);
+  Value scaleDim = builder.create<arith::MulFOp>(loc, dimFp, scaleFactorsFp);
+  Value floorDim = builder.create<math::FloorOp>(loc, scaleDim);
+  Value scaledDimToIndex = castIntToIndex(
+      builder, loc,
+      builder.create<arith::FPToSIOp>(loc, dimInt.getType(), floorDim));
+
+  return {scaleFactorInt, scaledDimToIndex};
+}
+
+// Returns the scaleFactor given input and output dimension.
+static Value getScaleFactor(OpBuilder &builder, Location loc, Value dim,
+                             int64_t scaledDim) {
+  Value scaledDimVal = builder.create<arith::ConstantOp>(
+      loc, builder.getI64IntegerAttr(scaledDim));
+  Value dimInt = castIndexToInt64(builder, loc, dim);
+  Value scaleFactorInt =
+      builder.create<arith::CeilDivSIOp>(loc, scaledDimVal, dimInt);
+  return scaleFactorInt;
+}
+
+// N, C, H, W = input_tensor.shape
+// N, C, H_scaled, W_scaled = out_tensor.shape
+// H_factor, W_factor = H_scaled/H, W_scaled/W
+
+// for i in range(N):
+//    for j in range(C):
+//      for k in range(H_scaled):
+//          for l in range(W_scaled):
+//              out_tensor[i, j, k, l] = input[i, j, k//H_factor, l//W_factor]
+
+namespace {
+class ConvertAtenUpsampleNearest2dVecOp
+    : public OpConversionPattern<AtenUpsampleNearest2dVecOp> {
+
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenUpsampleNearest2dVecOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Location loc = op->getLoc();
+    Value input = adaptor.input();
+
+    Type resultType = getTypeConverter()->convertType(op.getResult().getType());
+
+    auto inputRank = input.getType().cast<RankedTensorType>().getRank();
+    Type elementType =
+        input.getType().cast<RankedTensorType>().getElementType();
+
+    SmallVector<Value> dims = getTensorSizes(rewriter, loc, input);
+    SmallVector<Value> scaleFactorsInt;
+
+    if (!adaptor.scale_factors().getType().isa<Torch::NoneType>()) {
+      SmallVector<double> scaleFactors;
+      if (!matchPattern(adaptor.scale_factors(),
```

Comment:
As far as I can tell, the lowering for this op does not need this to be a constant float list. It seems like all the floats are converted back to values. You should use `getListConstructElements` instead.